pong/game/game.py

```diff
@@ -28,69 +16,78 @@ def __init__(self):
 
-    def init(self):
+    def new(self):
         pygame.init()
         pygame.font.init()
-        self.client = Client()
-        self.client.connect_server()
         self.running = True
         self.surface = pygame.display.set_mode(self.size)
-        pygame.display.set_caption("Pong")
         self.sprites = pygame.sprite.Group()
-
-        if self.client.is_host:
-            self.player1 = Player(self, 10, HEIGHT//2)
-            self.player2 = Player(self, WIDTH-10, HEIGHT//2)
-        else:
-            self.player1 = Player(self, WIDTH-10, HEIGHT//2)
-            self.player2 = Player(self, 10, HEIGHT//2)
-
+        self.menu_widgets = pygame.sprite.Group()
+        pygame.display.set_caption("Pong")
         self.ball = Ball(self)
+        self.player1 = Player(self)
+        self.player2 = Player(self)
         self.score = [0, 0]
-        
+        self.menu = Menu(self)
+
+        # client
+        self.client = Client(self, connect=False)
+        #self.client.set_players_positions()
         
     def cleanup(self):
         pygame.font.quit()
         pygame.quit()
+        sys.exit(2)
 
     def render(self):
         self.sprites.draw(self.surface)
         draw_text(self.surface, f'{self.score[0]} x {self.score[1]}',
          (255,255,255), WIDTH//2-5, 0, 32)
 
     def loop(self):
-        pygame.display.set_caption(f"Pong - FPS: {round(self.clock.get_fps(), 1)}")
-        positions = self.client.send_position(self.player1.pos, self.ball.pos, self.score)
-        if self.client.is_host:
-            resp = self.ball.update()
-            if resp == 2 and self.ball.x_dir < 0:
-                self.score[1]+=1
-            elif resp == 2 and self.ball.x_dir > 0:
-                self.score[0]+=1
-            print(self.score)
-        else:
-            self.ball.set_pos(*positions.ball_pos)
-            self.score = positions.score
-            
-        self.player2.update(*positions.other_player_pos)
+        self.client.update()
         self.player1.move()
-        
 
     def event(self, event):
         if event.type == pygame.QUIT:
             self.running = False
 
     def execute(self):
-        self.init()
+        self.new()
         while(self.running):
             for event in pygame.event.get():
                 self.event(event)
             self.surface.fill((0,0,0))
+            pygame.display.set_caption(f"Pong - FPS: {round(self.clock.get_fps(), 1)}")
             if self.client.is_playable:
                 self.loop()
                 self.render()
             else:
-                show_waiting_window(self.surface)
+                self.show_waiting_window()
             pygame.display.flip()
             self.clock.tick(60)
         self.cleanup()
+
+    def menu_screen(self):
+        while True:
+            for event in pygame.event.get():
+                self.menu.handle_event(event)
+            self.menu_widgets.draw(self.surface)
+            pygame.display.flip()
```

Comment:
Essa chamada vai bloquear a função execute francisco, já que tem uma while true.

Reply:
Sim, mas só por enquanto.
Quando for implementado os callbacks dos botões ele vai sair desse while
Ainda tô implementando